pulser/sequence.py

```diff
@@ -1111,6 +1128,45 @@ def _reset_parametrized(self) -> None:
         self._variables = {}
         self._to_build_calls = []
 
+    def config_slm_mask(self, qubits: Set[QubitId]) -> None:
+        """Setup an SLM mask by specifying the qubits it targets."""
+        if self._slm_mask_targets:
+            raise ValueError("SLM mask can be configured only once")
+
+        try:
+            targets = set(qubits)
+        except TypeError:
+            raise TypeError("The SLM targets should be castable to set")
+
+        if not targets.issubset(self._qids):
+            raise ValueError("SLM mask targets must exist in the register")
+
+        if not self._in_xy and self._channels:
+            raise NotImplementedError("SLM mask can only be added in XY mode")
+
+        # If checks have passed, set the SLM mask targets
+        self._slm_mask_targets = targets
+
+        # Find tentative initial and final time of SLM mask if possible
+        for channel in self._channels:
+            # Cycle on slots in schedule until the first pulse is found
+            for slot in self._schedule[channel]:
+                if isinstance(slot.type, Pulse):
+                    first = slot
+                    break
+            try:
+                ti = first.ti
+                tf = first.tf
+            # If try fails, it's because the variable 'first' is not defined,
+            # meaning there are no pulses in this channel
+            except NameError:
+                continue
+            try:
+                if ti < self._slm_mask_time[0]:
+                    self._slm_mask_time = [ti, tf]
+            except IndexError:
+                self._slm_mask_time = [ti, tf]
```

Comment:
Instead of breaking the for-loop right away, why don't you make these checks once you find a first pulse? Also, it's often considered a good practice to avoid using `try-except` statements and to "move the code to the left" as much as possible. Combining all this together would look something like this:
```suggestion
            for slot in self._schedule[channel]:
                if not isinstance(slot.type, Pulse):
                    continue
                ti = slot.ti
                tf = slot.tf
                if self._slm_mask_time:
                    if ti < self._slm_mask_time[0]:
                        self._slm_mask_time = [ti, tf]
                else:
                    self._slm_mask_time = [ti, tf]
                break
```

Reply:
looks good